src/main/java/io/pravega/connectors/flink/util/FlinkPravegaUtils.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.connectors.flink.util;
+
+import io.pravega.connectors.flink.EventTimeOrderingOperator;
+import io.pravega.connectors.flink.FlinkPravegaWriter;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.datastream.DataStreamSink;
+
+public class FlinkPravegaUtils {
+
+    private FlinkPravegaUtils() {
+    }
+
+    /**
+     * Writes a stream of elements to a Pravega stream with event time ordering.
+     * <p>
+     * This method returns a sink that automatically reorders events by their event timestamp.  The ordering
+     * is preserved by Pravega using the associated routing key.
+     *
+     * @param stream      the stream to read.
+     * @param writer      the Pravega writer to use.
+     * @param parallelism the degree of parallelism for the writer.
+     * @param <T>         The type of the event.
+     * @return a sink.
+     */
+    public static <T> DataStreamSink<T> writeToPravegaInEventTimeOrder(DataStream<T> stream, FlinkPravegaWriter<T> writer, int parallelism) {
```

Comment:
@StephanEwen does this seem legit to you, to force a forwarding strategy.   Striving to ensure that all elements for a given key are written by a specific sink instance.

Reply:
Note that the Kafka producer has a similarly complex `writeTo` method that composes an internal operator with the sink operator.   For different reasons.  Just pointing out the similarity.
https://github.com/apache/flink/blob/release-1.3.2/flink-connectors/flink-connector-kafka-0.10/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer010.java#L118